include/flexflow/ops/linear_params.h

```diff
@@ -0,0 +1,80 @@
+#ifndef _FLEXFLOW_LINEAR_PARAMS_H
+#define _FLEXFLOW_LINEAR_PARAMS_H
+
+#include "flexflow/ffconst.h"
+#include "flexflow/fftype.h"
+#include "flexflow/parallel_tensor.h"
+
+namespace FlexFlow {
+
+class LinearParams {
+public:
+  LayerID layer_guid;
+  int out_channels;
+  bool use_bias;
+  DataType data_type;
+  ActiMode activation;
+
+  bool is_valid(ParallelTensorShape const &input_shape) const;
+  void solve_dims(const ParallelTensor input,
```

Comment:
Out of scope, but because `ParallelTensor` is just a `ParallelTensorBase *`, `ParallelTensor const` is actually a `const` pointer to a non-`const` `ParallelTensorBase`, which I'm pretty sure is not the desired behavior (correct me if I'm wrong here @jiazhihao). We should really fix this at some point, probably by defining a proper `ParallelTensor` containing a `std::shared_ptr<ParallelTensorBase>`

Reply:
Tracking in #302